SetupDataPkg/Tools/ConfigEditor.py

```diff
@@ -793,19 +798,24 @@ def load_from_delta(self):
         self.load_delta_file(path)
 
     def load_delta_file(self, path):
-        # assumption is that there is only one yaml file and one delta file
-        # while is applied to that yaml file and one xml file and one csv applied
-        # to that xml file
+        # assumption is there is only one yaml file (deprecated) but there may be multiple xml files
+        # so we can only load this delta file if the file name matches to this xml data
         file_id = -1
         yml_id = -1
         xml_id = -1
         is_variable_list_format = True
         for idx in self.cfg_data_list:
+            # if we have a yaml, fall back to old behavior
+            # if loading xml, ensure that the file path we have matches the csv
+            # we are trying to load
             if self.cfg_data_list[idx].config_type == 'yml':
                 yml_id = idx
-            else:
+            elif re.search(self.cfg_data_list[idx].cfg_data_obj._cur_page + ".csv", path) is not None:

```

Comment:
nit: will there ever be an "else" case? is the non-handling a correct action?

Reply:
Line 816 is the else case, taking the whole list into consideration, not just that index. I.e. we go through the list of loaded configs to see if this csv filename matches any loaded configs. If it doesn't, then xml_id stays at -1 and line 817 will reaise an exception.